Userland/Libraries/LibWeb/HTML/WindowOrWorkerGlobalScope.cpp

```diff
@@ -86,6 +90,28 @@ WebIDL::ExceptionOr<String> WindowOrWorkerGlobalScopeMixin::atob(String const& d
     return TRY_OR_THROW_OOM(vm, decoder->to_utf8(decoded_data.value()));
 }
 
+// https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-queuemicrotask
+void WindowOrWorkerGlobalScopeMixin::queue_microtask(WebIDL::CallbackType& callback)
+{
+    auto& vm = this_impl().vm();
+    auto& realm = *vm.current_realm();
+
+    JS::GCPtr<DOM::Document> document;
+    if (is<Window>(this_impl())) {
+        document = &static_cast<Window&>(this_impl()).associated_document();
+    } else {
+        // FIXME: Support getting to the document from within a worker
+        return;
```

Comment:
I think we should just pass a null document to `queue_a_microtask` for workers. :)